cypress/integration/home_page_test.js

```diff
@@ -0,0 +1,99 @@
+describe('Home page tests', () => {
+  context('no access before login', () => {
+    it('redirects to login', function () {
+      cy.visit('/');
+      cy.location('pathname').should('eq', '/login');
+    });
+  });
+
+  context('authenticated', () => {
+    beforeEach(() => {
+      cy.fixture('devices_stats').as('devicesStats');
+      cy.fixture('interfaces').as('interfaces');
+      cy.fixture('test.astarte.FirstInterface').as('interface');
+      cy.fixture('interface_majors').as('interfaceMajors');
+      cy.fixture('triggers').as('triggers');
+      cy.server();
+      cy.route('GET', '/appengine/v1/*/stats/devices', '@devicesStats');
+      cy.route('GET', '/realmmanagement/v1/*/triggers', '@triggers');
+      cy.route('GET', '/realmmanagement/v1/*/interfaces', '@interfaces');
+      cy.route('GET', '/realmmanagement/v1/*/interfaces/*', '@interfaceMajors');
+      cy.route('GET', '/realmmanagement/v1/*/interfaces/*/*', '@interface');
+      cy.route('GET', '/appengine/health', '');
+      cy.route('GET', '/realmmanagement/health', '');
+      cy.route('GET', '/pairing/health', '');
+      cy.route('GET', '/flow/health', '');
+      cy.login();
+    });
+
+    it('successfully loads the Home page after logging in', () => {
+      cy.location('pathname').should('eq', '/');
+      cy.get('h2').contains('Astarte Dashboard');
+    });
+
+    it('has an API Status card with a table to display services health', () => {
+      cy.get('.card')
+        .contains('API Status')
+        .parents('.card')
+        .within(() => {
+          const services = ['Realm Management', 'AppEngine', 'Pairing', 'Flow'];
+          cy.get('table tbody').find('tr').should('have.length', services.length);
+          services.forEach((service, index) => {
+            cy.get(`table tbody tr:nth-child(${index + 1})`).within(() => {
+              cy.contains(service);
+              cy.contains('This service is operating normally');
+            });
+          });
+        });
+    });
+
+    it('has a Devices card that shows stats on devices', function () {
+      cy.get('.card')
+        .contains('Devices')
+        .parents('.card')
+        .within(() => {
+          cy.contains('Registered devices').next().contains(this.devicesStats.data.total_devices);
+          cy.contains('Connected devices')
+            .next()
+            .contains(this.devicesStats.data.connected_devices);
+        });
+    });
+
+    it('has a Interfaces card that shows available interfaces', function () {
+      cy.get('.card')
+        .contains('Interfaces')
+        .parents('.card')
+        .within(() => {
+          this.interfaces.data.forEach((interfaceName) => {
```

Comment:
this should not be a for-each. This card displays a maximum of 5 interfaces

Reply:
Nice catch!